patches/server/0928-Add-entity-knockback-API.patch

```diff
@@ -0,0 +1,21 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: MelnCat <melncatuwu@gmail.com>
+Date: Sun, 16 Oct 2022 12:10:17 -0700
+Subject: [PATCH] Add entity knockback API
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+index e807ef287136e7b3931197e45434a3f618cf3054..152b07ad1d1027d5dd5cd861a687229c04853c66 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+@@ -977,5 +977,10 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
+         }
+         throw new IllegalArgumentException(entityCategory + " is an unrecognized entity category");
+     }
++
++    @Override
++    public void knockback(double strength, double directionX, double directionZ) {
```

Comment:
This should add a precondition if the strength is negative instead of ignoring it

Reply:
Precondition added

Reply:
I mean the strength must be positive (excluding zero): strength > 0.0D

Reply:
Ah, alright